application/src/main/java/org/togetherjava/tjbot/commands/moderation/modmail/ModMailCommand.java

```diff
@@ -0,0 +1,184 @@
+            .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+            .build();
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        long userId = event.getUser().getIdLong();
+
+        if (handleIsOnCooldown(userId, event)) {
+            return;
+        }
+        authorToLastModMailInvocation.put(userId, Instant.now());
+        event.deferReply().setEphemeral(true).queue();
+
+        long userGuildId = event.getOption(OPTION_GUILD).getAsLong();
+        Optional<TextChannel> modMailAuditLog = getModMailChannel(event.getJDA(), userGuildId);
+        if (modMailAuditLog.isEmpty()) {
+            logger.warn(
+                    "Cannot find the designated modmail channel in server by id {} with the pattern modmail",
+                    userGuildId);
+            return;
+        }
+
+        MessageCreateAction message = createMessage(event, userId, modMailAuditLog.orElseThrow());
+
+        sendMessage(event, message);
+
+    }
+
+    private boolean handleIsOnCooldown(long userId, SlashCommandInteractionEvent event) {
+        if (isChannelOnCooldown(userId)) {
+            event.reply("Can only be used once per %s minutes.".formatted(COOLDOWN_DURATION_VALUE))
+                .setEphemeral(true)
+                .queue();
+            return true;
+        }
+        return false;
+    }
+
+    private Optional<TextChannel> getModMailChannel(JDA jda, long userGuildI) {
+        return jda.getGuildById(userGuildI)
+            .getTextChannelCache()
+            .stream()
+            .filter(channel -> modMailChannelNamePredicate.test(channel.getName()))
+            .findAny();
+    }
+
+    private MessageCreateAction createMessage(SlashCommandInteractionEvent event, long userId,
+            TextChannel modMailAuditLog) {
+        String userMessage = event.getOption(OPTION_MESSAGE).getAsString();
+        boolean wantsToStayAnonymous = event.getOption(OPTION_STAY_ANONYMOUS).getAsBoolean();
+        String user;
+        MessageCreateAction message;
+        if (wantsToStayAnonymous) {
+            user = "Anonymous";
+            message = modMailAuditLog.sendMessageEmbeds(createModMailMessage(user, userMessage));
+        } else {
+            user = event.getUser().getAsMention();
+            message = modMailAuditLog.sendMessageEmbeds(createModMailMessage(user, userMessage))
+                .addActionRow(DiscordClientAction.General.USER.asLinkButton("Author Profile",
+                        String.valueOf(userId)));
+        }
+        return message;
+    }
+
+    private void sendMessage(SlashCommandInteractionEvent event, MessageCreateAction message) {
+        InteractionHook hook = event.getHook();
+        message.mapToResult()
+            .map(result -> result.isSuccess() ? "Your message has been forwarded, thanks."
+                    : "There was an issue forwarding your message, sorry. We are investigating.")
```

Comment:
u need to do the log message here. since u did `mapToResult()`, errors from this part do not flow into the `error -> ` thing at the bottom anymore. ur error handler at the bottom only gets issues with the `editOriginal` (which u dont have to cover)

Reply:
Done